.kokoro/presubmit/downstream-compatibility-spring.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is only meant to test spring-cloud-gpc, specifically on its
+# autoconfig generation workflow
+
+set -eox pipefail
+
+# Get the directory of the build script
+scriptDir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
+cd "${scriptDir}/../.." # cd to the root of this repo
+source "$scriptDir/common.sh"
+
+install_repo_modules
+GAPIC_GENERATOR_VERSION=$(parse_pom_version "gapic-generator-java-bom")
+echo "Install complete. gapic-generator-java-bom = $GAPIC_GENERATOR_VERSION"
+
+pushd gapic-generator-java/target
+# Download and configure spring-cloud-gcp for testing
+last_release=$(find_last_release_version "spring-cloud-gcp" "main" "GoogleCloudPlatform")
+git clone "https://github.com/GoogleCloudPlatform/spring-cloud-gcp.git" --depth=1 --branch "v$last_release"
+update_all_poms_dependency "spring-cloud-gcp" "gapic-generator-java-bom" "${GAPIC_GENERATOR_VERSION}"
```

Comment:
@lqiu96 This `update_all_poms_dependency` looks like a nice util that maybe we can reuse in testing protoc.

Reply:
The script I wrote to update the poms is pretty much a copy of this implementation. When I originally wrote it, I had the limitation where i couldn't use xmllint or xmlstartlet in kokoro.